engine/src/communication/CommunicationInterface/bufferTransport.cpp

```diff
@@ -178,11 +178,13 @@ void buffer_transport::send(const char * buffer, size_t buffer_size){
 
 
 void buffer_transport::increment_frame_transmission() {
+	std::unique_lock<std::mutex> lock(mutex);
```

Comment:
these are atomics they shouldnt need this lock

Reply:
according to https://en.cppreference.com/w/cpp/thread/condition_variable it is still needed though
```
Even if the shared variable is atomic, it must be modified under the mutex in order to correctly publish the modification to the waiting thread.
```

That said, these locks did not seem to affect the behavior